checks/linux.py

```diff
@@ -0,0 +1,176 @@
+from .vars import *
+from .utils.utils import *
+
+
+def getSessionTypeLine(lines):
+    sessionType = search('Session Type:', lines)
+    if len(sessionType) > 0:
+        return sessionType[0]
+
+
+def getWindowSystemLine(lines):
+    windowSystem = search('Window System:', lines)
+    if len(windowSystem) > 0:
+        return windowSystem[0]
+
+
+def checkDistro(lines):
+    isDistroNix = search('Distribution:', lines)
+
+    if len(isDistroNix) <= 0:
+        return
+
+    distro = isDistroNix[0].split()
+    distro = distro[2:]
+    distro = ' '.join(distro)
+
+    return [LEVEL_INFO, distro, ""]
+
+
+def checkFlatpak(lines):
+    isFlatpak = search('Flatpak Runtime:', lines)
+
+    if len(isFlatpak) > 0:
+        return [LEVEL_INFO, "Flatpak",
+                "You are using the Flatpak. Plugins are available as Flatpak extensions, which you can find in your Distribution's Software Center or via <code>flatpak search com.obsproject.Studio</code>. Installation of external plugins is not supported."]
+
+
+def checkSnapPackage(lines):
+    isDistroNix = search('Distribution:', lines)
+
+    if len(isDistroNix) <= 0:
+        return
+
+    distro = isDistroNix[0].split()
+    # Snap Package logs "Ubuntu Core" as distro, so it gets split halfway
+    if distro[2] == '"Ubuntu' and distro[3] == 'Core"':
```

Comment:
I realise this is not the PR that introduced this, but relying on split() and precise array indices is not highly reliable, and more importantly, will crash the whole analyzer if split() returns less than 4 elements (which would happen in case of for example a log straight from sdtout so without timestamps).
I'd instead advocate for `if  'Ubuntu Core' in isDistroNix[0]:`

Reply:
> and more importantly, will crash the whole analyzer if split() returns less than 4 elements (which would happen in case of for example a log straight from sdtout so without timestamps).

I tested this, and no:
```
 cat stdout.log | grep "Distribution"
info: Distribution: "Ubuntu Core"
^0    ^1            ^2      ^3
 ./loganalyzer.py -f stdout.log | grep "Ubuntu"
Info:     Beta OBS Version (31.0.0-beta2), No Output Session, Third-Party Plugins (1), "Ubuntu Core", X11, KDE (KDE), VCam not available
"Ubuntu Core"
```

On `stdout` there's an `info:` prefix in place of the timestamp, so the index count stays the same.

Reply:
Fixed